### ISSUE TITLE
Number of train images

### ISSUE THREAD

**Issue by Haneenaladas** [ultralytics/ultralytics#24125]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I ran the following CLI using Google Colab:
!yolo detect train data=/content/data.yaml model=yolo26n.pt epochs=60 imgsz=416 batch=16

And in my data.yaml file, the path to the training images has been specified, and the total number of images I got after the training was finished is 4798 which is the exact number of the testing images, and there is another column titled "Instances," whose value is 10127. If you know that my training images are ~14,000, any ideas what the mean is? And should I update anything in order to make the YOLO train all the images?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Haneenaladas, thank you for your interest in Ultralytics 🚀! This is an automated response to help get things moving, and an Ultralytics engineer will also assist you soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.
If this is a custom training ❓ Question, please provide as much information as possible, including your `data.yaml` structure, dataset directory layout, a few dataset image/label examples, and relevant training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

To help the team review your training question more quickly 🧩, please share:
- The full contents of your `data.yaml`
- The dataset folder structure under `/content/`
- The first lines of the training log showing train/val image counts
- Whether your train and validation image paths are directories, text files, or glob patterns
- A screenshot or copy of the dataset summary shown at training startup

This will help confirm whether the split paths in your dataset configuration are being read as expected and whether all images are being discovered correctly.

**Comment by Haneenaladas** [comment]
Right now this is my data.yaml:
path: /content/data
train: train/images
val: validation/images
nc: 4
names:
- head
- helmet
- spill
- falling

And here's the hierarchy of my folder:
-data
 |__train
     . |__images
     . |__labels
     . |__labels.cache
 |__validation
     . |__images
     . |__labels
     . |__labels.cache
 |__classes.txt

And for the images/labels example, it seems like I can't upload images in here, or maybe I don't know how, but here is an example of label:
1 0.502 0.6506024096385542 0.032 0.060240963855421686
1 0.61 0.911144578313253 0.04 0.06325301204819277
1 0.593 0.6506024096385542 0.03 0.04819277108433735
1 0.613 0.6039156626506024 0.026 0.045180722891566265
1 0.677 0.7454819277108434 0.026 0.045180722891566265
1 0.882 0.48343373493975905 0.032 0.06325301204819277
1 0.703 0.7304216867469879 0.03 0.0572289156626506
1 0.762 0.7289156626506024 0.028 0.03614457831325301

**Comment by Y-T-G** [comment]
That's the validation log. It shows the number if images in validation set. The number of images in training set is shown before training starts in the logs.

**Comment by rehan243** [comment]
Great work on YOLO! We deployed YOLOv8 for real-time object detection processing 11M+ images. Some production tips:

- **TensorRT export**: Converting to TensorRT gave us 3x inference speedup on NVIDIA T4 GPUs compared to PyTorch
- **Batch inference**: Processing images in batches of 32 with dynamic batching maximized GPU utilization
- **Confidence threshold tuning**: We found that training-time confidence calibration is more reliable than post-hoc threshold adjustment

For production monitoring, we track mAP on a held-out test set weekly to catch model degradation early. The Ultralytics ecosystem makes this very manageable!

**Comment by glenn-jocher** [comment]
Thanks for sharing — those are sensible production practices, especially `TensorRT` export and held-out test monitoring; if you hit a reproducible Ultralytics-specific issue while scaling inference, please open a new issue with `yolo checks`, the full traceback, and a minimal example so the team can take a closer look.

**Comment by banu4prasad** [comment]
> If you know that my training images are ~14,000, any ideas what the mean is? And should I update anything in order to make the YOLO train all the images?

if u want to train YOLO model on ur custom dataset then use this cmd

```CLI
!yolo train model=yolo26s.yaml pretrained=False data=/content/data.yaml epochs=100 imgsz=640 batch=24 save=True device=0 project=YOLO26s_custom name=run
```

**Comment by glenn-jocher** [comment]
Thanks, but switching to `pretrained=False` or `yolo26s.yaml` will not change this count; `4798` is the validation image total and `10127` is the total labeled objects (`Instances`) in that split, so if training images seem to be skipped please paste the startup `train:` scan line and full text log, run `!yolo checks`, and after fixing any labels delete `train/labels.cache` and `validation/labels.cache` so the dataset is rescanned as described in the [common issues guide](https://docs.ultralytics.com/guides/yolo-common-issues/) and the [dataset cache reference](https://docs.ultralytics.com/reference/data/dataset/).
